test/parallel/test-buffer-regression-649.js

```diff
@@ -6,8 +6,14 @@ const SlowBuffer = require('buffer').SlowBuffer;
 
 // Regression test for https://github.com/nodejs/node/issues/649.
 const len = 1422561062959;
-assert.throws(() => Buffer(len).toString('utf8'));
-assert.throws(() => SlowBuffer(len).toString('utf8'));
-assert.throws(() => Buffer.alloc(len).toString('utf8'));
-assert.throws(() => Buffer.allocUnsafe(len).toString('utf8'));
-assert.throws(() => Buffer.allocUnsafeSlow(len).toString('utf8'));
+const lenLimitMsg = /^RangeError: Invalid typed array length$/;
+assert.throws(() => Buffer(len).toString('utf8'),
+              lenLimitMsg);
```

Comment:
Addressed.

A noob question: is it better that I amend the commits myself during the PR so you don't have to squash them when you land them, or is it better I just leave several commits in the PR so the review process won't get hidden in GitHub?